File: python/leetcode/347.py
nums = [1, 1, 1, 2, 2, 3]

k = 2


def topKFrequent(nums, k):
    dic = {}
    answer = []

    for num in nums:
        if num in dic:
            dic[num] += 1
        else:
            dic[num] = 1
    dic = {k: v for k, v in sorted(dic.items(), key=lambda item: -item[1])[:k]}

    for k, v in dic.items():
        answer.append(k)

    return answer


assert topKFrequent(nums, k) == [1, 2]
assert topKFrequent([1], 1) == [1]

File: javascript/leetcode/347.js
const nums = [1, 1, 1, 2, 2, 3];
const k = 2;

let topFrequent = function (nums, k) {
  let map = {};
  const answer = [];

  // counting
  for (let num of nums) {
    if (num in map) {
      map[num] += 1;
    } else {
      map[num] = 1;
    }
  }

  let sortedMap = Object.entries(map)
    .sort(([, a], [, b]) => a - b)
    .reverse()
    .slice(0, k);

  //   console.log(sortedMap);
  for (let item of sortedMap) {
    answer.push(parseInt(item[0]));
  }
  return answer;
};
